riscv/src/compiler.rs

```diff
@@ -444,7 +444,14 @@ fn preamble<T: FieldElement>(degree: u64, runtime: &Runtime, with_bootloader: bo
         "".to_string()
     };
 
-    let mul_instruction = mul_instruction::<T>();
+    for machine in ["binary", "shift"] {
```

Comment:
```suggestion
    for machine in ["binary", "shift", "split_gl"] {
```
it's needed for the `mul` instruction

Reply:
ah you added it below already